src/benchmark.rs

```diff
@@ -173,3 +199,59 @@ fn fetch_configured_refs(repo: &git2::Repository, refs: &[String]) -> Result<Pat
     }
     Ok(wd)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use core::panic;
+    #[tokio::test]
+    async fn test_resolve_env() {
+        let resolved_envs =
+            resolve_env_from_stdout(Command::new("echo").arg("[\"env0\", \"env1\", \"env2\"]"))
+                .await;
+        match resolved_envs {
+            Ok(v) => {
+                assert_eq!(v[0], "env0");
+                assert_eq!(v[1], "env1");
+                assert_eq!(v[1], "env1");
+            }
+            _ => panic!("Parsing unexpectedly failed for echo-ing a list of strings."),
+        }
+    }
```

Comment:
You can use `Result` in tests, making them more concise.

Also you checked `v[1]` twice and `v[2]` never, so I recommend comparing vecs directly.

```suggestion
    async fn test_resolve_env() -> Result<()> {
        let resolved_envs =
            resolve_env_from_stdout(Command::new("echo").arg("[\"env0\", \"env1\", \"env2\"]"))
                .await?;
        assert_eq!(resolved_envs, vec!["env0", "env1", "env2"]);
    }
```

Reply:
right! totally forgot about `expect`!